packages/shared/src/types.ts

```diff
@@ -35,7 +35,7 @@ type RouteConfig = {
 export type FileRouterInputKey = AllowedFileType | MimeType;
 
 export type ExpandedRouteConfig = Partial<
-  Record<FileRouterInputKey, RouteConfig>
+  {[key in FileRouterInputKey]: RouteConfig}
 >;
```

Comment:
Nit but i think this is more kind on the compiler

```suggestion
export type ExpandedRouteConfig = {
  [key in FileRouterInputKey]?: RouteConfig
};

Reply:
nvm that caused the same error wtf... ts is weeeeird